rdt/transformers/base.py

```diff
@@ -193,15 +194,17 @@ def _add_columns_to_data(data, columns, column_names):
         Returns:
             ``pandas.DataFrame`` with the new columns added.
         """
-        if columns is not None:
-            if isinstance(columns, (pd.DataFrame, pd.Series)):
-                columns.index = data.index
+        if isinstance(transformed_data, (pd.Series, np.ndarray)):
+            transformed_data = pd.DataFrame(transformed_data, columns=transformed_names)
 
-            if len(columns.shape) == 1:
-                data[column_names[0]] = columns
-            else:
-                new_data = pd.DataFrame(columns, columns=column_names)
-                data = pd.concat([data, new_data.set_index(data.index)], axis=1)
+        if transformed_names:
+            transformed_data.index = data.index
```

Comment:
should we copy `transformed_data` since we are modifying its index or does it not matter?

Reply:
shouldn't matter. I simplified this part of the code either way